pandas/tests/generic/test_metadata.py

```diff
@@ -0,0 +1,49 @@
+import pytest
+
+import pandas as pd
+from pandas.core._meta import PandasMetadata
+
+mymeta = PandasMetadata("attr")
+
+
+@mymeta.register(pd.core.generic.NDFrame.copy)
+def _(new, other):
+    new.attr = other.attr + 1
+
+
+@mymeta.register("concat")
```

Comment:
an alternative to this registering which might be more self-contained is subclassing PandasMetadata and implementing/overriding the relevant methods/functions

Reply:
Hmm, yes... I'll think about this. This may fix the note I had in the module docstring that we don't document pandas calls actually call `__finalize__`, and what they're called with.